app/routes/forum/categories/category/threads/thread/posts/index.js

```diff
@@ -68,6 +68,24 @@ export default class PostIndexRoute extends AuthenticatedRoute {
     this.controller.send('goToNextPage');
   }
 
+  @action
+  beforeModel(transition) {
+    const routeName = 'forum.categories.category.threads.thread.posts.index';
+    // check that transition is not from the same thread
+    if (
+      !(
+        transition.from?.name === routeName &&
+        transition.from?.attributes?.thread?.id ===
+          this.modelFor('forum.categories.category.threads.thread')?.id
+      )
+    ) {
+      const controller = this.controllerFor(routeName);
+      if (controller) {
+        controller.send('resetNewContent');
+      }
+    }
+  }
+
```

Comment:
haven't touched this, will have a look at whether this is octane style

Reply:
I've simplified this.
see https://api.emberjs.com/ember/4.8/classes/Route/methods/resetController?anchor=resetController